### ISSUE TITLE
brie-factor utility causes error 

### ISSUE THREAD

**Issue by linkerst** [huangyh09/brie#4]
I am following the instructions of the brie manual ([(https://brie-rna.sourceforge.io/manual.html)]). At step 4 I run into the following error:

`brie-factor -a AS_events/SE.gold.gtf -r GRCm38.p5.genome.fa -c mm10.60way.phastCons.bw -o mouse_features.csv -p 10`

loading annotation file... Done.
extracting features for 5227 skipping exon triplets with 10 cores...
Traceback (most recent call last):
  File "/myhome /conda-envs/ml_0/bin/brie-factor", line 11, in <module>
    sys.exit(main())
  File "/myhome /conda-envs/ml_0/lib/python2.7/site-packages/brie/brie_factor.py", line 152, in main
    RV = result[g].get()
  File "/myhome /conda-envs/ml_0/lib/python2.7/multiprocessing/pool.py", line 567, in get
    raise self._value
OSError: [Errno 2] No such file or directory 

I receive this error both with my and the example data. I treated the bigWigSummary utility as described in the manual and my multiprocessing packages should be fine. 
The first error message looks like the program has trouble finding the relative imports `from .utils.gtf_utils import loadgene` . 

Could you please help me finding the issue?

Thanks 
Stephanie 

P.S
The code in step 3 
`brie-event-filter -a AS_events/SE.gff3 -anno_ref gencode.vM12.annotation.gtf -r GRCm38.p5.genome.fa`

should be changed to
`brie-event-filter -a AS_events/SE.gff3 --anno_ref gencode.vM12.annotation.gtf -r GRCm38.p5.genome.fa`

**Comment by huangyh09** [comment]
Hi Stephanie,

Thanks for using BRIE, and apologize for the delay. I just tested the brie-factor on my machine and it works well, but I can't direct see the problem in your case. 

First, I don't think it is the problem with ``from .utils.gtf_utils import loadgene`` as the loading annotation file can be finished. 

Second, could you try a few things more to diagnose the bug:
- run ``brie-factor -a AS_events/SE.gold.gtf -r GRCm38.p5.genome.fa -c mm10.60way.phastCons.bw -o mouse_features.csv -p 1`` this will use only one CUP so it will avoid the error from multiprocessing.
- run ``bigWigSummary`` to see whether you could run it directly.

Could you let me if you try these two things? 

Best,
Yuanhua

**Comment by linkerst** [comment]
Dear Yuanhua,

Thanks for your reply. 

- Running BRIE with the –p option leads to the same error I posted above. 
- Running bigWigSummary directly in terminal works, e.g running `bigWigSummary hg38.phyloP100way.bw  chrX 4000 80000 1` has the output 0.0755242. 

Best,
Stephanie

**Comment by huangyh09** [comment]
Hi Stephanie,

Sorry I haven't figured out this error. I suspect this is because of the incompatibility between the python2.7/multiprocessing/pool.py and the system or some other Python packages. I found quite a few similar errors report by [searching your error report](https://www.google.co.uk/?gws_rd=ssl#newwindow=1&q=lib/python2.7/multiprocessing/pool.py%22,+line+567,+in+get+raise+self._value+OSError:+[Errno+2]+No+such+file+or+directory). You may get some ideas of the bug.

In addition, it is strange to see the same error when you tried ``-p 1`` option, as this mode won't use multiprocessing any more. 

Let me know if you have any idea of the bug.

Best,
Yuanhua

**Comment by linkerst** [comment]
Hi Yuanhua,

I think I managed to fix my problem. 

First I created a new conda environment just for BRIE because sometimes I have issues with version conflicts. Afterward I was able to run brie-factor with the `–p 1` option without using multiprocessing. 

However, I run into a new bug concerning subprocessing.  Adding   `shell=True` to the subprocessing command in your get_factor function in the fasta_utils.py file (line 130) solved that problem und brie-factor run successfully. 

Best,
Stephanie
